src/main/java/io/gravitee/policy/aws/lambda/AwsLambdaPolicy.java

```diff
@@ -33,4 +51,90 @@ public AwsLambdaPolicy(AwsLambdaPolicyConfiguration configuration) {
     public String id() {
         return "aws-lambda";
     }
+
+    @Override
+    public Completable onRequest(HttpPlainExecutionContext ctx) {
+        return invokeAndHandleLambda(ctx)
+            .flatMapCompletable(invokeResponse -> {
+                if (configuration.isSendToConsumer()) {
+                    ctx.setInternalAttribute(
+                        InternalContextAttributes.ATTR_INTERNAL_INVOKER,
+                        new LambdaInvoker(
+                            configuration,
+                            ctx.getInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_INVOKER),
+                            invokeResponse
+                        )
+                    );
+                }
+                return Completable.complete();
+            });
+    }
+
+    @Override
+    public Completable onResponse(HttpPlainExecutionContext ctx) {
+        return invokeAndHandleLambda(ctx)
+            .flatMapCompletable(invokeResponse -> {
+                if (configuration.isSendToConsumer()) {
+                    Buffer buffer = Buffer.buffer(invokeResponse.payload().asByteArray());
+                    ctx.response().headers().remove(CONTENT_LENGTH);
+                    ctx.response().status(invokeResponse.statusCode());
+                    return ctx.response().onChunks(chunks -> chunks.ignoreElements().andThen(Flowable.just(buffer)));
+                }
+                return Completable.complete();
+            });
+    }
+
+    private <T extends HttpBaseExecutionContext> Single<InvokeResponse> invokeAndHandleLambda(T ctx) {
+        return Single
+            .fromFuture(invokeLambda(ctx.getTemplateEngine()))
+            .subscribeOn(Schedulers.io())
+            .flatMap(invokeResponse -> {
+                log.debug("AWS Lambda function has been invoked successfully");
+                if ("Handled".equals(invokeResponse.functionError()) || "Unhandled".equals(invokeResponse.functionError())) {
+                    return processError(ctx, null, AwsLambdaError.AWS_LAMBDA_INVALID_RESPONSE);
+                } else if (invokeResponse.statusCode() >= 200 && invokeResponse.statusCode() < 300) {
+                    return processSuccess(ctx, invokeResponse);
+                }
+                return processError(ctx, null, AWS_LAMBDA_INVALID_STATUS_CODE);
+            })
+            .onErrorResumeNext(throwable -> {
+                log.error("An error occurs while invoking lambda function", throwable);
```

Comment:
I don't think we should log the exception here. I would prefer having a way to propagate the exception in the `ExecutionFailure` instead, but this is not supported at the moment, so, at least, I would not log it as an error but as debug, directly in the processError method. WDYT?